synapse/http/client.py

```diff
@@ -726,6 +733,14 @@ async def get_file(
                 HTTPStatus.BAD_GATEWAY, "Got error %d" % (response.code,), Codes.UNKNOWN
             )
 
+        if is_allowed_content_type and b"Content-Type" in resp_headers:
+            content_type = resp_headers[b"Content-Type"][0]
+            if not is_allowed_content_type(content_type):
+                raise SynapseError(
+                    HTTPStatus.BAD_GATEWAY,
+                    ("Unsupported content type for URL previews: %r" % content_type),
```

Comment:
I don't think it is great to leak the `b` prefix in our error messages:

```suggestion
                    ("Unsupported content type for URL previews: %s" % (content_type.decode(errors='ignore'),)),
```

This kind of leaks that we're using this only for URL previews, but not sure there's a better thing to do besides just letting the passed in callable raise.

Reply:
> This kind of leaks that we're using this only for URL previews, but not sure there's a better thing to do besides just letting the passed in callable raise.

Yeah, now that you pointed it out, it sticks out as ugly. I think I'll just say "Requested file's content type not allowed for this operation" and live with the less specific error message. Looking the calling contexts of `get_file`, the operation being performed should be obvious from the surrounding log lines anyway.

Reply:
Sounds perfect. Thank you!